src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
+        this->actor.textId = Text_GetFaceReaction(globalCtx, 5);
+        this->unk_30B = 1;
+        this->unk_300 = 6;
+        this->actionFunc = func_80A5475C;
+        return;
+    }
+    if (((gSaveContext.eventChkInf[0] & 0x200) != 0) && ((gSaveContext.eventChkInf[2] & 0x20) != 0)) {
+        if ((gSaveContext.eventChkInf[3] & 0x80) != 0) {
+            // "Get all spiritual stones!"
+            osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 全部の精霊石GET！ ☆☆☆☆☆ \n" VT_RST);
+            this->unk_300 = 6;
+            this->actor.textId = 0x7006; // "There's a lot going on in the castle right now. I can't allow even..."
+            this->actionFunc = func_80A5475C;
+            return;
+        }
+    }
+    if (gSaveContext.nightFlag != 0) {
+        // "Sleep early for children!"
+        osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ 子供ははやくネロ！ ☆☆☆☆☆ \n" VT_RST);
+        this->unk_300 = 6;
+        this->actor.textId = 0x7002; // "Welcome to Hyrule Castle Town. It's a peaceful, prosperous town."
+        this->actionFunc = func_80A5475C;
+        return;
+    }
+    if (this->unk_30C != 0) {
+        // "Anything passes"
+        osSyncPrintf(VT_FGCOL(BLUE) " ☆☆☆☆☆ なんでも通るよ ☆☆☆☆☆ \n" VT_RST);
+        this->unk_300 = 6;
+        this->actor.textId = 0x7099; // "KEEP IT A SECRET FROM EVERYONE"
+        this->actionFunc = func_80A5475C;
+        return;
+    }
+    if ((gSaveContext.eventChkInf[1] & 4) != 0) {
+        if (this->unk_30E == 0) {
+            // "Start under the first sleeve!"
+            osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ １回目袖の下開始！ ☆☆☆☆☆ \n" VT_RST);
+            this->actor.textId = 0x7071; // "Do you really want to go through this gate that much?..."
+            this->unk_30E = 1;
+        } else {
+            // "Start under the second sleeve!"
+            osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ ２回目袖の下開始！ ☆☆☆☆☆ \n" VT_RST);
+            this->actor.textId = 0x7072; //  "Give me 10 Rupees...Yes...No"
+        }
+        this->unk_300 = 4;
+        this->actionFunc = func_80A5475C;
+
+    }
+
+    else {
+        // "That's okay"
+        osSyncPrintf(VT_FGCOL(CYAN) " ☆☆☆☆☆ それはとおらんよぉ ☆☆☆☆☆ \n" VT_RST);
+        this->unk_300 = 6;
+        this->actor.textId = 0x7029; // "So you say you want to see Princess Zelda, eh?..."
+        this->actionFunc = func_80A5475C;
+    }
+}
+
+void func_80A5344C(EnHeishi2* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx) != 0)) {
+
+        this->unk_300 = 5;
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0:
+                if (gSaveContext.rupees >= 10) {
+                    Rupees_ChangeBy(-10);
+                    thisx->textId = 0x7098; // "OK, I'll let you pass, quickly!"
+                    this->actionFunc = func_80A53538;
+                    break;
+                }
+
+                else {
```

Comment:
else here should be on the same line as the closing brace for the if. I think cland would catch this so make sure to run it